From e7f406f968408c19fc28c8bdaaf14fe35667a79c Mon Sep 17 00:00:00 2001
From: Ruslan Ruslichenko <ruslan.ruslichenko@globallogic.com>
Date: Wed, 18 Feb 2015 10:11:53 -0800
Subject: [PATCH 090/111] ARM: OMAP4: Enable sg chaining support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Scatter list should always fit to single physical page, which
limits max number of chunks it can represent.
This causes some problems, for example, with ion heaps:

1. Allocation memory from ion system heap sometimes hits BUG_ON
in __sg_alloc_table when memory is fragmented and number of chunks
is larger than SG_MAX_SINGLE_ALLOC.
2. Tiler 2D buffers almost always exceed SG_MAX_SINGLE_ALLOC limit,
which make them imposible to map to single scatter/gather list.

Enabling SG chaining support allow to remove this limitation, but ARM
still don't enable it by default and SG chaining should be enabled per SoC.
As for OMAP looks like all drivers updated and ready to use new SG list API,
so there should be no problems to enable it.

Change-Id: I6e7a558e62a80559286492cc8cc02404f66dc7c1
Signed-off-by: Ruslan Ruslichenko <ruslan.ruslichenko@globallogic.com>
Signed-off-by: D. Andrei Măceș <dmaces@nd.edu>
---
 arch/arm/mach-omap2/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-omap2/Kconfig b/arch/arm/mach-omap2/Kconfig
index 8341d37..69b2b7c 100644
--- a/arch/arm/mach-omap2/Kconfig
+++ b/arch/arm/mach-omap2/Kconfig
@@ -54,6 +54,7 @@ config ARCH_OMAP4
 	select PM_OPP if PM
 	select USB_ARCH_HAS_EHCI
 	select ARCH_HAS_BARRIERS
+	select ARM_HAS_SG_CHAIN
 
 comment "OMAP Core Type"
 	depends on ARCH_OMAP2
-- 
2.7.4

